src/main/kotlin/org/rust/ide/inspections/RsLint.kt

```diff
@@ -14,14 +17,15 @@ import org.rust.lang.core.psi.ext.*
 enum class RsLint(
     val id: String,
     val groupIds: List<String> = emptyList(),
-    val defaultLevel: RsLintLevel = RsLintLevel.WARN
+    val defaultLevel: RsLintLevel = WARN,
+    val highlightType: Map<RsLintLevel, ProblemHighlightType> = mapOf(WARN to WARNING, DENY to GENERIC_ERROR)
 ) {
     NonSnakeCase("non_snake_case", listOf("bad_style")),
     NonCamelCaseTypes("non_camel_case_types", listOf("bad_style")),
     NonUpperCaseGlobals("non_upper_case_globals", listOf("bad_style")),
-    Deprecated("deprecated"),
-    UnusedVariables("unused_variables", listOf("unused")),
-    NeedlessLifetimes("clippy::needless_lifetimes", listOf("clippy::complexity", "clippy::all", "clippy"));
+    Deprecated("deprecated", highlightType = mapOf(WARN to LIKE_DEPRECATED, DENY to GENERIC_ERROR)),
```

Comment:
Personally, I don't like this mapping via `map` because it's quite easy to forget some items in mapping and get wrong highlighting.
I suggest implementing a bit more wordy but less error-prone way. Let's add something like
```
protected open fun toHighlightingType(level: RsLintLevel): ProblemHighlightType {
    return when (level) {
        ALLOW -> INFORMATION
        WARN -> WARNING
        DENY, FORBID -> GENERIC_ERROR
    }
}
```
and override it for each lint variant (if necessary, of course). For example, for `Deprecated`:
```
override fun toHighlightingType(level: RsLintLevel): ProblemHighlightType {
    return if (level == WARN) LIKE_DEPRECATED else super.toHighlightingType(level)
}
```

Of course, you will have to move `getProblemHighlightType` from `RsLintInspection` to `RsLint` but it's ok

Reply:
> you will have to move `getProblemHighlightType` from `RsLintInspection` to `RsLint` but it's ok

Why do I need to move `getProblemHighlightType`?

Reply:
Because `toHighlightingType` is protected and can be used only from `RsLint`. I don't think we should make `toHighlightingType` public because it's an implementation detail of `RsLint`

Reply:
Oh, it seems that I misunderstood, I thought `toHighlightingType` was declared in `RsLintInspection`. 😆